### DISCUSSION TITLE
Apply trim function to all (unknown) columns in select statement

### DISCUSSION THREAD

**Discussion by darthf1** [duckdb/duckdb#8022] [General]
I would like to convert an Excel file to CSV, via [Excel import](https://duckdb.org/docs/guides/import/excel_import), but all contents must be trimmed.

I do not know the column names of the Excel file upfront. Is there a way to apply a function to all columns, without defining them?

```
COPY (SELECT * FROM str_read('my_excel.xlsx') TO 'result.csv' (FORMAT 'csv', HEADER, DELIMITER ',')
```
Also, what happens if I do not provide a `layer` in this example? Does the first sheet get selected?

**Comment by Tishj** [comment]
I believe that's what the `COLUMNS(*)` expression is for
<https://duckdb.org/docs/sql/expressions/star>

**Reply by darthf1** [reply]
Thanks. If I read it correctly, with `COLUMNS(*)` I can do some logic as to which column I want to include in the result, but it does not explain how I can apply a function to each column, without defining/knowing all column names?

**Reply by Tishj** [reply]
```sql
D select length(COLUMNS(*)) from (VALUES ('abc', 'def', 'also a string')) tbl(a, b, c);
┌───────────────┬───────────────┬───────────────┐
│ length(tbl.a) │ length(tbl.b) │ length(tbl.c) │
│     int64     │     int64     │     int64     │
├───────────────┼───────────────┼───────────────┤
│             3 │             3 │            13 │
└───────────────┴───────────────┴───────────────┘
```

**Reply by darthf1** [reply]
Ah yeah that is exactly it. Very nice! Thanks!

If I can iterate on that; would you happen to know how to get the original column name as an alias? So instead of 
`length(tbl.a), length(tbl.b), length(tbl.c)` you would get `a, b, c` as headers.

**Reply by Tishj** [reply]
I believe that that is less straight forward still, it was asked before:
<https://github.com/duckdb/duckdb/discussions/7898>

Maybe the answer there is also useful for you

**Reply by darthf1** [reply]
Ok, many thanks :D
